packages/indexer/src/indexer.ts

```diff
@@ -94,16 +103,17 @@ export function createIndexer<TFilter, TBlock, TRet>({
 export async function run<TFilter, TBlock, TRet>(
   client: Client<TFilter, TBlock>,
   indexer: Indexer<TFilter, TBlock, TRet>,
+  sink?: Sink<TRet> | Promise<Sink<TRet>>,
 ) {
   await indexerAsyncContext.callAsync({}, async () => {
     await indexer.hooks.callHook("run:before");
 
-    const sink = (await indexer.options.sink) ?? defaultSink();
+    const _sink = (await sink) ?? defaultSink();
```

Comment:
Don't name variables with `_` unless it's a variable we want to ignore. You can change the parameter name to something like `sinkPromise` or `sinkArg`.